llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -10227,21 +10227,25 @@ void SelectionDAGBuilder::visitVACopy(const CallInst &I) {
 }
 
 SDValue SelectionDAGBuilder::lowerRangeToAssertZExt(SelectionDAG &DAG,
-                                                    const Instruction &I,
```

Comment:
This should not lose the ability to handle range metadata. We should still be able to lower loads etc. with range to include the assert

Reply:
ok will fix, only noticed that it was called for calls now and it made the code a bit simpler.

Reply:
this is fixed now